src/frontend/components/DashboardVideoLivePairing/index.tsx

```diff
@@ -0,0 +1,139 @@
+import { Box, Meter, Text } from 'grommet';
+import React, { useEffect, useState } from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+
+import { DashboardButton } from 'components/DashboardPaneButtons/DashboardButtons';
+import { usePairingVideo } from 'data/queries';
+import { Video } from 'types/tracks';
+
+const messages = defineMessages({
+  pairingSecretLabel: {
+    defaultMessage: 'Pair an external device',
+    description:
+      'Button label to request a secret for pairing an external device.',
+    id: 'components.DashboardVideoLivePairing.pairingSecretLabel',
+  },
+  pairingSecretDisplay: {
+    defaultMessage: 'Pairing secret: {secret}',
+    description:
+      'Message displaying the secret for pairing an external device.',
+    id: 'components.DashboardVideoLivePairing.pairingSecretDisplay',
+  },
+  pairingSecretCountdownExpired: {
+    defaultMessage: 'Pairing secret expired',
+    description:
+      'Message displayed when the secret for pairing an external device has expired.',
+    id: 'components.DashboardVideoLivePairing.pairingSecretCountdownExpired',
+  },
+});
+
+interface DashboardVideoLivePairingProps {
+  video: Video;
+}
+
+export const DashboardVideoLivePairing = ({
+  video,
+}: DashboardVideoLivePairingProps) => {
+  const [buttonColor, setButtonColor] = useState('brand');
+  const [secret, setSecret] = useState('');
+  const [expiresIn, setExpiresIn] = useState(0);
+  const [secondsLeft, setSecondsLeft] = useState(0);
+  const [percentsLeft, setPercentsLeft] = useState(100);
+  const [expired, setExpired] = useState(false);
+
+  const usePairingVideoMutation = usePairingVideo(video.id, {
+    onSuccess: (data) => {
+      setSecret(data.secret);
+      setExpiresIn(data.expires_in);
+      setSecondsLeft(data.expires_in);
+      setPercentsLeft(100);
+      setExpired(false);
+      setButtonColor('status-ok');
+    },
+  });
+
+  const pairVideoAction = () => {
+    usePairingVideoMutation.mutate();
+  };
+
+  const onInterval = () => {
+    setSecondsLeft(secondsLeft - 1);
+    setPercentsLeft((secondsLeft * 100) / expiresIn);
+
+    if (secondsLeft <= 0) {
+      if (secondsLeft > -3) {
+        setExpired(true);
+        setButtonColor('status-warning');
+      } else {
+        setSecret('');
+        setButtonColor('brand');
+      }
+    }
+  };
+
+  useEffect(() => {
+    let interval: number | undefined;
+    if (secret) {
+      interval = window.setInterval(onInterval, 1000);
```

Comment:
In fact this is not a `setInterval` you want. But a `setTimeout`. Because every time `secondsLeft` changes, a new `setInterval` is created and it is not the purpose of this function.